.travis.yml

```diff
@@ -18,32 +29,207 @@ python:
   before_deploy: false
   deploy: false
 
-stages:
-  - name: sentinel
-    if: branch != master OR type = pull_request
-  - test
-  - moban
-  - name: unsupported
-    if: branch = master AND type = push
-
-.check_moban: &check_moban
+.moban: &moban
   <<: *disable_global
   python: 3.6
   stage: moban
-  install: pip install moban
-  script: .ci/check_moban.sh
-  if: branch != master OR type = pull_request
+  install: pip install moban>=0.0.4
+  script:
+    - moban
+    - git diff --exit-code
 
 jobs:
   include:
-    - stage: sentinel
-      # All other jobs will be cancelled if the sentinel job fails
-      <<: *disable_global
-      python: 3.6
-      install: pip install -r requirements.txt ".[alldeps]"
-      script: coala --non-interactive -V
+    # Manually added "language" entries should complete test coverage
+    - stage: test-languages
+      language: lua
+      env: BEARS=lua
+      addons:
+        apt:
+          packages:
+            - luarocks
+
+    - stage: test-languages
+      language: generic
+      env: BEARS=opam INFER_VERSION=0.7.0
+           PATH=$PATH:$HOME/infer-linux64-v$INFER_VERSION/infer/bin
+      addons:
+        apt:
+          sources:
+            - avsm
+          packages:
+            - camlp4-extra
+            - ocaml
+            - opam
+
+    - stage: test-languages
+      # This is in generic image, as language: haskell isnt working yet
+      # https://github.com/coala/coala-bears/issues/1384
+      language: generic
+      dist: xenial
+      env: DIST=xenial BEARS=adhoc BEAR_LIST="ghc-mod hlint"
+      # ghc-mod needs parts of ghc, specifically at least /usr/lib/ghc/settings
+      # ghc-mod needs cabal-install
+      # shellcheck is failing in xenial
+      addons:
+        apt:
+          packages:
+            - cabal-install
+            - ghc
+            - ghc-mod
+            - hlint
+
+    # BEARS=apt_get rather than 'apt' to match key in bear-metadata.yaml
+    - stage: test-languages
+      language: generic
+      dist: xenial
+      env: DIST=xenial BEARS=apt_get DISABLE_BEARS=shellcheck
+      addons:
+        apt:
+          packages:
+            - chktex
+            - cppcheck
+            - devscripts
+            - flawfinder
+            - indent
+            - libperl-critic-perl
+            - libxml2-utils
+            - mono-mcs
+            - php7.0-cli
+            - phpmd
+            - php-codesniffer
+            - verilator
+
+    - stage: test-languages
+      language: generic
+      dist: xenial
+      env: DIST=xenial BEARS=adhoc BEAR_LIST="bakalint default-jre"
```

Comment:
this should be merged with the above entry using

env: DIST=xenial BEARS=adhoc BEAR_LIST="apt_get bakalint default-jre" DISABLE_BEARS="shellcheck .."

merging with `ghc-mod` isnt good, as that should be moved to a new `language: haskell` , but hlint doesnt require ghc, so it could be added to the main xenial job.